File: 4-curso/2 cuatrimestre/DS/Practica 4/DS-master/practica4/DS-master/practica4/barcos/lib/ForoPesca/FiltroHora.dart
import 'Filtro.dart';
import 'package:intl/intl.dart';


class FiltroHora implements Filtro {
  late DateTime _fecha;
  late String _fechaDia;

  FiltroHora() {
    _fecha = DateTime.now();

    _fechaDia = DateFormat('yyyy-MM-dd').format(_fecha);
  }

  @override
  String ejecutar(String texto) {
    String resultado = '$texto ${_fechaDia.toString()}';
    return resultado;
  }

  @override
  String toString(){
    return "Soy un filtro hora: ${_fechaDia.toString()}\n";
  }
}

File: 4-curso/2 cuatrimestre/DS/Practica 1/Ejercicios/Ejercicios 2/p1/ejercicio2/src/main/java/com/p1/FiltroHora.java
package com.p1;

import java.util.Date;

public class FiltroHora implements Filtro{
    private Date fecha;

    FiltroHora(){
        fecha = new Date();
    }

    @Override
    public String ejecutar(String texto){
        String texto_resultado = texto + " " + fecha.toString(); //Objeto fecha creado
        return texto_resultado;
    }
}